extensions/cache/src/main/java/org/radargun/stages/mapreduce/MapReduceStage.java

```diff
@@ -290,31 +288,28 @@ public DistStageAck executeOnSlave() {
    }
 
    private DistStageAck executeMapReduceTask(Task<KOut, VOut, R> task, Statistics stats, boolean supportsResultCacheName) {
-      long durationNanos;
-      long start;
-
       MapReduceAck ack = new MapReduceAck(slaveState);
       try {
          if (collatorFqn != null) {
-            start = TimeService.nanoTime();
+            Request request = stats.startRequest();
             payloadObject = task.executeWithCollator();
-            durationNanos = TimeService.nanoTime() - start;
-            stats.registerRequest(durationNanos, MapReducer.MAPREDUCE_COLLATOR);
+            // TODO handle exception
```

Comment:
Matej is about to update MR stages - there was no exception handling and I haven't added one, but marked this place as missing it... I've focused just on the API change.